api/types/src/convert.rs

```diff
@@ -21,30 +21,35 @@ use move_binary_format::file_format::FunctionHandleIndex;
 use move_core_types::{
     identifier::Identifier,
     language_storage::{ModuleId, StructTag},
-    resolver::MoveResolver,
 };
 use move_resource_viewer::MoveValueAnnotator;
 
 use crate::transaction::{ModuleBundlePayload, StateCheckpointTransaction};
 use anyhow::{ensure, format_err, Result};
+use aptos_state_view::StateView;
 use aptos_types::state_store::state_key::StateKey;
+use aptos_vm::data_cache::{IntoMoveResolver, RemoteStorageOwned};
 use serde_json::Value;
 use std::{
     convert::{TryFrom, TryInto},
     rc::Rc,
 };
 
-pub struct MoveConverter<'a, R: ?Sized> {
-    inner: MoveValueAnnotator<'a, R>,
+pub struct MoveConverter<S> {
+    resolver: RemoteStorageOwned<S>,
 }
 
-impl<'a, R: MoveResolver + ?Sized> MoveConverter<'a, R> {
-    pub fn new(inner: &'a R) -> Self {
+impl<S: StateView> MoveConverter<S> {
+    pub fn new(state_view: S) -> Self {
         Self {
-            inner: MoveValueAnnotator::new(inner),
+            resolver: state_view.into_move_resolver(),
         }
     }
 
+    pub fn annotator(&self) -> MoveValueAnnotator<RemoteStorageOwned<S>> {
+        MoveValueAnnotator::new(&self.resolver)
```

Comment:
this is very expensive, there's a module cache lives inside the annotator

Reply:
The `MoveConverter` is reconstructed for every call anyway..  but what's the module you used to do internal refs again?

Reply:
ouroboros, but I think I tried it doesn't work with async context. there's Rc inside

Reply:
Okay it's ouroboros.. and it's tricky to get compiled..  I'll land de hopefully follow up later. Thank you.